### PR TITLE
Only clear cache for frozen tier shards

### PR DESCRIPTION
**Pull request by ywelsch**
Makes it so that the experimental searchable snapshot clear cache API only clears the shared cache (`FrozenCacheService`), leaving `CacheService` unchanged.

This API is only useful to measure how much shared_cache a query might take (take cache stats, reset the cache, then run the query, the check cache stats again). Running this API against the cold tier would dramatically mess with query performance there, as it undoes the pre-warming, so this PR removes that effect.

### CODE REVIEW

File: x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/store/SearchableSnapshotDirectory.java [hunk 1]
```diff
@@ -356,7 +356,6 @@ public final void close() {
     public void clearCache() {
```
Comment (tlrx) [inline x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/store/SearchableSnapshotDirectory.java:356]: I'd change this to `clearCaches(boolean cold, boolean frozen)` and uses it also in tests

File: x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/store/SearchableSnapshotDirectoryTests.java [hunk 2]
```diff
@@ -786,7 +787,12 @@ public void testClearCache() throws Exception {
                     }
                     assertListOfFiles(cacheDir, allOf(greaterThan(0), lessThanOrEqualTo(nbRandomFiles)), greaterThan(0L));
                     if (randomBoolean()) {
-                        directory.clearCache();
+                        directory.clearCache(); // should not clear CacheService cache
+                        assertBusy(() -> assertListOfFiles(cacheDir, greaterThan(0), greaterThan(0L)));
```
Comment (tlrx) [inline x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/store/SearchableSnapshotDirectoryTests.java:791]: See previous comment, I'd use `directory.clearCaches(true, true)` here and avoid an extra assertBusy

File: x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/store/SearchableSnapshotDirectory.java (hunk 1 above)
Comment (ywelsch) [reply x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/store/SearchableSnapshotDirectory.java]: good idea, I've changed it according to your suggestion.

**Review by tlrx — APPROVED** [review]
LGTM

**Pull request merged**